services/test-validators.js

```diff
@@ -50,11 +50,11 @@ const isPhpVersionReduction = withRegex(/^((>= \d+(\.\d+)?)|(\d+\.\d+(, \d+\.\d+
 const isStarRating = withRegex(/^(?=.{5}$)(\u2605{0,5}[\u00BC\u00BD\u00BE]?\u2606{0,5})$/);
 
 // Required to be > 0, because accepting zero masks many problems.
```

Comment:
Could you update this comment too.

Reply:
Hmm.. Now you've pointed out this comment, I've gone and looked back at the history. I probably should have done this before submitting this PR:

It looks like this was introduced in response to a particular problem, where an error condition was manifesting as a zero value: https://github.com/badges/shields/pull/1201

That seems like a bit of an edge case, but I now wonder if there are other cases like this I'm not thinking of :thinking: 

So maybe would be better to leave the regexes requiring a non-zero value and just switch this test to a more active repo. Then I tried to find one and realised that while the badge is correctly reporting the number from the CocoaPods API, CocoaPods seems to be incorrectly reporting all (or many) packages as having zero downloads per week. For example, this is one of the most popular packages on cocoapods: http://cocoapods.org/pods/Alamofire its unlikely that this has really had >300,000 downloads this month with 0 of them in the last week. This seems to be a [reported issue](https://github.com/CocoaPods/cocoapods.org/issues/348) which has been open for some time :/

Reply:
Would we be better to show an error message rather than `0`?
Although it won't always be valid, neither will `0` and if it is actually `0`, it would be less likely to go noticed that the message is incorrect.
![](https://img.shields.io/:downloads-unavailable-lightgrey.svg)

Reply:
@chris48s good investigation. https://github.com/badges/shields/pull/1201 was a really ugly bug. 
@RedSparr0w `0` is OK for me, better than `unavailable`. Stats are available and are equal `0`. 

In my opinion this kind of bugs can be avoided when we have tests which stubs and verifies exact value. 
What do you think about replacing tests for weekly downloads and weekly apps with?
```js
t.create('downloads (valid, weekly)')
  .get('/dw/AFNetworking.json')
  .intercept(nock => nock('https://metrics.cocoapods.org')
    .get('/api/v1/pods/AFNetworking')
    .reply(200, { stats: { download_week: 2 }}))
  .expectJSON({
    name: 'downloads',
    value: '2/week'
  });

t.create('apps (valid, weekly)')
  .get('/aw/AFNetworking.json')
  .intercept(nock => nock('https://metrics.cocoapods.org')
    .get('/api/v1/pods/AFNetworking')
    .reply(200, { stats: { app_week: 2 }}))
  .expectJSON({
    name: 'apps',
    value: '2/week'
  });
```